api/client/proxy.go

```diff
@@ -25,22 +25,44 @@ import (
 	"net/url"
 
 	"github.com/gravitational/trace"
-	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/proxy"
 )
 
-// DialProxy creates a connection to a server via an HTTP Proxy.
+// DialProxy creates a connection to a server via an HTTP or SOCKS5 Proxy.
 func DialProxy(ctx context.Context, proxyURL *url.URL, addr string) (net.Conn, error) {
 	return DialProxyWithDialer(ctx, proxyURL, addr, &net.Dialer{})
 }
 
-// DialProxyWithDialer creates a connection to a server via an HTTP Proxy using a specified dialer.
-func DialProxyWithDialer(ctx context.Context, proxyURL *url.URL, addr string, dialer ContextDialer) (net.Conn, error) {
+// DialProxyWithDialer creates a connection to a server via an HTTP or SOCKS5 Proxy using a specified dialer.
+func DialProxyWithDialer(
+	ctx context.Context,
+	proxyURL *url.URL,
+	addr string,
+	dialer *net.Dialer,
+) (net.Conn, error) {
 	if proxyURL == nil {
 		return nil, trace.BadParameter("missing proxy url")
 	}
+
+	switch proxyURL.Scheme {
+	case "http":
+		return dialProxyWithHTTPDialer(ctx, proxyURL, addr, dialer)
```

Comment:
nit: Could you return wrapped with `trace.Wrap(err)` that will provide proper error stack stack trace. 
```suggestion
		conn, err :=  dialProxyWithHTTPDialer(ctx, proxyURL, addr, dialer)
		return conn, trace.Wrap(err) 
```